lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -251,6 +241,34 @@ def call(env)
           @app.call(env)
         end
 
+        def store_password(env, domain_config, job_id)
+          server_job_result = env[:cloudstack_compute].query_async_job_result({:jobid => job_id})
+          if server_job_result.nil?
+            env[:ui].warn(' -- Failed to retrieve job_result for retrieving the password')
+            return
+          end
+
+          while true
```

Comment:
Check, better to timeout then an endless loop. Copied this from `create_port_forwarding_rule` and `create_firewall_rule` which can be improved as well :wink: Do we have a general cloud communication timeout value somewhere or use a specific timeout/parameter for each different operation